### DISCUSSION TITLE
Check the connection status

### DISCUSSION THREAD

**Discussion by fbettcba** [redisson/redisson#5184] [Q&A] [answered]
Is there any way to get the connection status? I mean, if the redisson client is currently connected or not. 
I saw some references to getNodesGroup and ping, but getNodesGroup is deprecated.

**Comment by uilton-oliveira** [comment]
That would be helpful... sometimes, randomly after we restart our platform daily, one of our nodes is unable to connect to redis for some unknown reason... we already tried logging in the container and connecting with Redis CLI and the connection works fine.

So, the only solution is killing this node and let it restart it... but we need a way to iterate over all connections used in this node to see if it's healthy or not, and if not for a given time, we could force a kill in the container...

**Comment by mrniko** [comment, marked as answer]
> I saw some references to getNodesGroup and ping, but getNodesGroup is deprecated.

Use `org.redisson.api.RedissonClient#getRedisNodes()` method instead. It has `pingAll()` method and `ping()` method per Redis node.
